### PR TITLE
Move hypothesis dependence to testing

### PR DESCRIPTION
**Pull request by jochym**
This is an attempt to break circular dependency on hypothesis

### CODE REVIEW

**Comment by CJ-Wright** [comment]
@jochym Feel free to add yourself to the maintainers if you want.

File: recipe/meta.yaml [hunk 1]
```diff
@@ -55,3 +55,4 @@ extra:
     - CJ-Wright
     - licode
     - nicoddemus
+    - jochym
```
Comment (CJ-Wright) [inline recipe/meta.yaml]: :+1:

**Comment by nicoddemus** [comment]
Thanks @jochym!

Actually I just realized that none of the dependencies in `requirements/run` [are needed](https://github.com/python-attrs/attrs/blob/master/setup.py#L94), they are all [test dependencies only](https://github.com/python-attrs/attrs/blob/master/dev-requirements.txt).

Would you mind moving the other dependencies as well? Also this is a perfect candidate for a `noarch` package, but I will ask for the bot to rerender for us.

**Comment by nicoddemus** [comment]
@conda-forge-admin, please add noarch: python

**Comment by nicoddemus** [comment]
> ... but I will ask for the bot to rerender for us.

What a time to be alive! 😄

**Comment by nicoddemus** [comment]
@jochym please also bump the build number. 👍

**Comment by jochym** [comment]
Merge?

**Comment by nicoddemus** [comment]
Awesome @jochym, thanks!

**Comment by jakirkham** [comment]
Unfortunately we need to revert the `noarch: python` conversion here.

xref: https://github.com/regro/cf-scripts/issues/264
xref: https://github.com/regro/cf-scripts/pull/265
xref: https://github.com/conda-forge/conda-smithy/issues/643
xref: https://github.com/conda-forge/requests-feedstock/issues/23

**Pull request merged**